### ISSUE TITLE
`no-expression-statements`: allow `await Promise<void>` when `ignoreVoid` is `true`

### ISSUE THREAD

**Issue by cn-stephen** [eslint-functional/eslint-plugin-functional#1052]
## Suggestion

Right now, thanks to https://github.com/eslint-functional/eslint-plugin-functional/issues/865 we don't get flagged by a `Promise<void>` returning function

```
groupService.delete(params.id); <-- not flagged
```

But if we `await` it, which we almost always should, it's flagged
```
await groupService.delete(params.id); <-- flagged :(
```

I think we should allow `await Promise<void>` for these scenarios, when we're using the `ignoreVoid` option

**Comment by cn-stephen** [comment]
Here's what I'm doing locally in the meantime:
```
diff --git a/lib/index.js b/lib/index.js
index 9e0a2dc28a7a9a0043e5762e0ab895d1e07ea65d..483e3a53ee6608f53c1342914eaf5b40e7417bdb 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -2317,8 +2317,12 @@ function checkExpressionStatement(node, context, options) {
         };
     }
     const { ignoreVoid, ignoreSelfReturning } = optionsObject;
-    if ((ignoreVoid || ignoreSelfReturning) && isCallExpression(node.expression)) {
-        const returnType = getTypeOfNode(node.expression, context);
+    // Unwrap AwaitExpression to get the inner call expression
+    const effectiveExpression = node.expression.type === "AwaitExpression" && node.expression.argument != null
+        ? node.expression.argument
+        : node.expression;
+    if ((ignoreVoid || ignoreSelfReturning) && isCallExpression(effectiveExpression)) {
+        const returnType = getTypeOfNode(effectiveExpression, context);
         if (returnType === null) {
             return {
                 context,
@@ -2337,7 +2341,7 @@ function checkExpressionStatement(node, context, options) {
             };
         }
         if (ignoreSelfReturning) {
-            const type = getTypeOfNode(node.expression.callee, context);
+            const type = getTypeOfNode(effectiveExpression.callee, context);
             if (type !== null) {
                 const declaration = type.getSymbol()?.valueDeclaration;
                 if (typescript !== undefined &&
```

**Comment by cn-stephen** [comment]
yay! Thank you!!
